corehq/apps/hqadmin/views/users.py

```diff
@@ -394,7 +394,11 @@ def form_valid(self, form):
 
         verb = 're-enabled' if self.user.is_active else 'disabled'
         reason = form.cleaned_data['reason']
-        log_model_change(self.request.user, self.user, f'User {verb}. Reason: "{reason}"')
+        couch_user = CouchUser.from_django_user(self.user)
+        log_user_change(None, couch_user, self.request.couch_user,
+                        changed_via=USER_CHANGE_VIA_WEB, message=f'User {verb}. Reason: "{reason}"',
+                        fields_changed={'is_active': self.user.is_active},
```

Comment:
should this include the password reset?

Reply:
We could. I assume we don't track that as of now and will be new?
We would get to a discussion of adding logging for other updates that we have not been tracking so far. The PRs so far are to just use the new model wherever we are currently tracking.

Not sure, but will password reset via admin isn't something that will be useful from UserHistory perspective?
similar discussion i think will happen for all updates that we want to track now.

Reply:
You've added fields in other places already e.g. 2FA.

Fine to leave it but perhaps we should start a doc or something to note things that aren't tracked. I don't think the fact that it's done via admin has any bearing considering this is meant to be an audit log.

Reply:
oh sorry, I thought it was a different view/update you were referring to on HQ. I missed that it was in the same view and I had missed adding it. Adding that as a part of the message `"Password reset" if reset_password` since we can't save the password as the `fields_changed`

Reply:
you could just put `fields['password'] = '***'`

Reply:
> You've added fields in other places already e.g. 2FA.

Sorry about the confusion there. The change in 
https://github.com/dimagi/commcare-hq/pull/29890/commits/452e6769174fdf55a42223ab9a64de880634f78b#diff-e0a7331ea888053cfc991192ff9122d31820052cf81b278b2cabc2aa8c95cab9R500 came while updating to use log_user_create and log the actions taken.

Thinking about it some more now, i think its better to 
1. keep `fields_changed['devices'] = []` because devices is actually a user property
2. remove `fields_changed['two_factor_auth_disabled_until'] = disable_until` and put that information in the message instead as `f"Disabled  for {disable_for_days} days"` 

i have been trying to limit fields_changed to only user properties and not a generic dictionary to story any changes.

Reply:
> you could just put fields['password'] = '***'

Ya. Does "Password reset" makes sense from UI perspective, seeing that as the change message instead of seeing `Password: ***`?

Reply:
> keep fields_changed['devices'] = [] because devices is actually a user property

The devices being cleared are not `user.devices`, they are 2fa devices which is completely different

> remove fields_changed['two_factor_auth_disabled_until'] = disable_until and put that information in the message instead as f"Disabled for {disable_for_days} days"

:+1: 

> Ya. Does "Password reset" makes sense from UI perspective, seeing that as the change message instead of seeing Password: ***?

:+1:

Reply:
thanks @snopoke 
Made sense to put all in messages then https://github.com/dimagi/commcare-hq/pull/29890/commits/da0ac8021f818aff984f11bda9703e19051bd452